File: 项目/qz_services/admin-api/library/util/page.go
package util

import "math"

type PageStrut struct{
    TotalNumber int `json:"total_number"`
    PageTotalNumber int  `json:"page_total_number"`
    PageSize int  `json:"page_size"`
    PageCurrent int  `json:"page_current"`
}

// 统一分页信息返回格式
func GetPage(page int, limit int, count int) PageStrut {
    var pageObj PageStrut

    var pages int
    if limit > 0 {
        pages = int(math.Ceil(float64(count) / float64(limit)))
    }

    pageObj.TotalNumber = count
    pageObj.PageSize = limit
    pageObj.PageCurrent = page
    pageObj.PageTotalNumber = pages

    return pageObj
}

// 获取分页offset
func GetPageOffset(page int, limit int) int {
    offset := (page - 1) * limit
    return offset
}

// 分页参数处理
func PageFormat(page int, def int) int {
    if page <= 0 {
        page = def
    }

    return page
}

File: 项目/qz_api_sales/extend/Util/Page.php
<?php

namespace Util;

class Page
{

    public  $pageSize = 10; //每页条数
    public  $page_total_number = 0; //总页数
    public  $firstrow = 0;
    private $default_result = ["page_total_number" => 0,"total_number" => 0,"page_size" => 0,"page_current" => 0];
    private $pageIndex = 1; //当前页数
    private $total_number = 0;//总条数

    public function __construct($pageIndex = 1,$pageSize = 10,$total_number = 0)
    {
        $this->pageIndex = $pageIndex <= 0 ? 1 : $pageIndex;
        $this->pageSize = $pageSize <= 0 ? 10 : $pageSize;
        $this->total_number = $total_number <= 0 ? 0 : $total_number;
        $this->firstrow = ($this->pageIndex-1)*$this->pageSize;
        $this->page_total_number = ceil($this->total_number/$this->pageSize);
    }

    public function show()
    {
        return [
            "page_total_number" => (int)$this->page_total_number,
            "total_number" => (int)$this->total_number,
            "page_size" => (int)$this->pageSize,
            "page_current" =>(int) $this->pageIndex
        ];
    }

    public function default_show()
    {
       return $this->default_result;
    }
}
